src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -250,21 +250,19 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
       modalRef.componentInstance.sessionToCopyCourseId = this.courseId;
 
       modalRef.result.then((result: CopySessionModalResult) => {
-        this.feedbackSessionsService.getFeedbackSession({
-          courseId: this.courseId,
-          feedbackSessionName: this.feedbackSessionName,
-          intent: Intent.FULL_DETAIL,
-        }).pipe(
-            switchMap((feedbackSession: FeedbackSession) =>
-                this.copyFeedbackSession(feedbackSession, result.newFeedbackSessionName, result.copyToCourseId)),
-        ).subscribe((createdSession: FeedbackSession) => {
-          this.navigationService.navigateWithSuccessMessage(this.router,
-              '/web/instructor/sessions/edit',
-              'The feedback session has been copied. Please modify settings/questions as necessary.',
-              { courseid: createdSession.courseId, fsname: createdSession.feedbackSessionName });
-        }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); });
-      }, () => {});
-    }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorToast(resp.error.message); });
+        this.failedToCopySessions = {};
+        const requestList: Observable<FeedbackSession>[] = this.createSessionCopyRequestsFromModal(
+            result, this.courseId, this.feedbackSessionName);
+        if (requestList.length === 1) {
+          this.copySingleSession(requestList[0]);
```

Comment:
Single session is just a special case for copying multiple session, is it necessary to have a separate method to handle it?

Reply:
The flow for single session is completely different because the user gets redirected to edit the session, and since this happens at more than one place I figured having a separate method would be cleaner.